subsys/modbus/modbus_core.c

```diff
@@ -358,6 +359,7 @@ int modbus_disable(const uint8_t iface)
 		LOG_ERR("Unknown MODBUS mode");
 	}
 
+	k_work_cancel_sync(&ctx->server_work, &work_sync);
```

Comment:
No commit message, what are the expectations of this change? It is very unlikely that there will be a pending work.

Reply:
If you run modbus and the call back modbus client init,  this should garantee that nothing get pending.

Reply:
You mean when an application periodically initializes/disables an interface? If so, please add this comment to the commit message.

Reply:
Yes correct. I need to switch from modbus line discipline to pipe serial and drive a mux. I have create some function to clean up the pipe serial already. Can we agree on adjust the commit message or you thing that drain the uart is wrong? It's not mandatory to have it. I will upload  back the patches

Reply:
Please drop first commit and update commit message.